pkg/controller/account/account_controller.go

```diff
@@ -226,7 +221,10 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		ccsRoleID, result, initErr = r.initializeNewCCSAccount(reqLogger, currentAcctInstance, awsSetupClient, adminAccessArn)
 		if initErr != nil {
 			// TODO: If we have recoverable results from above, how do we allow them to requeue if state is failed
-			r.setStateFailed(reqLogger, currentAcctInstance, initErr.Error())
+			stateErr := r.setStateFailed(reqLogger, currentAcctInstance, initErr.Error())
+			if stateErr != nil {
+				reqLogger.Error(stateErr, "failed setting account state", "desiredState", "Failed")
+			}
```

Comment:
Totally optional:

These changes (throughout) will result in a pretty chatty log, since `setStateFailed` already logs. So perhaps we should just ignore the error (with or without using `//nolint`) and add a comment to explain why. E.g.
```suggestion
			// Ignore error: setStateFailed logs, and we want to return initErr rather than the error from setStateFailed.
			_ = r.setStateFailed(reqLogger, currentAcctInstance, initErr.Error())
```